### ISSUE TITLE
Isolate test fixtures

### ISSUE THREAD

**Issue by OldSneerJaw** [Kashoo/synctos#252]
Currently the test-helper module's state is shared (and reset) between all of a project's test cases. Adopt a pattern that produces a unique and isolated test fixture for each caller. In accordance with the current test helper implementation, the new implementation should expose two options to initialize the test fixture: from a document validation function file (e.g. `initFromValidationFunction`) or from a document definitions file (e.g. `initFromDocumentDefinitions`).

For example, a caller might initialize a test fixture like so:

```
const testFixtureMaker = require(synctos).testFixtureMaker;

describe('My document definitions:', () => {
  let testFixture;
  beforeEach(() => {
    testFixture = testFixtureMaker.initFromDocumentDefinitions('src/my-doc-definitions.js');
  });

  // ...
});
```

Be sure to [deprecate](https://nodejs.org/dist/latest-v8.x/docs/api/util.html#util_util_deprecate_function_string) the test-helper module and update examples in the README to account for the new style of instantiating a test fixture.

**Comment by dkichler** [comment]
Satisfied by #258

**Comment by OldSneerJaw** [comment]
I have identified some additional refactoring that should be done before this can be considered complete.

**Comment by dkichler** [comment]
Addressed by #261
